### PR TITLE
docs(Spaces): Add details for custom Python Spaces

### PR DESCRIPTION
**Pull request by ronvoluted**
Without making a full section on custom Python Spaces as it's still only unofficially supported, point out the main issues that are not obvious.

### CODE REVIEW

File: docs/hub/spaces.md
Comment (julien-c) [inline docs/hub/spaces.md:180]: can you qualify in which cases this is needed? (i haven't needed it yet on my side) like if you want a fluid vertical layout, etc?

File: docs/hub/spaces.md
Comment (julien-c) [inline docs/hub/spaces.md:186]: maybe put this one before the iframe resizer as it seems more generally applicable?

**Review by julien-c — COMMENTED** [review]
LGTM but pinging @cbensimon @christophe-rannou  on this!

File: docs/hub/spaces.md
Comment (ronvoluted) [reply docs/hub/spaces.md]: 👍 Have changed wording to be more clear about why it's needed

And that's right, needed essentially whenever the content exceeds the viewport height.

I just created a duplicate of one of the templates here as example, where the only change is iFrame Resizer being commented out: https://huggingface.co/spaces/ronvolutional/iframe-test
Compared to: https://huggingface.co/spaces/ronvolutional/http-server

 I tested Gradio HTML ouput and static SDK space before and they do not need it.

File: docs/hub/spaces.md
Comment (ronvoluted) [reply docs/hub/spaces.md]: Done. Originally it was because iFrame Resizer is essential (unless you have a short page) but not all Spaces have links.

File: docs/hub/spaces.md
Comment (julien-c) [inline docs/hub/spaces.md:186]:
```suggestion
Usually, the height of Spaces is adjusted automatically when using the Gradio library interface. If you provide your own frontend in the Gradio SDK however, and if your content height is larger than the viewport, you'll need to add an [iFrame Resizer script](https://cdnjs.com/libraries/iframe-resizer) so that content can scroll in the iframe:
```

File: docs/hub/spaces.md
Comment (julien-c) [inline docs/hub/spaces.md:191]:
```suggestion
As an example, here are two Spaces where the only change is iFrame Resizer being commented out: https://huggingface.co/spaces/ronvolutional/iframe-test
Compared to: https://huggingface.co/spaces/ronvolutional/http-server
```

File: docs/hub/spaces.md
Comment (julien-c) [reply docs/hub/spaces.md]: makes sense, and i think we can even link to your example Spaces, those are a great explanation

Comment (julien-c) [reply docs/hub/spaces.md]: (made some suggestions)

Comment (ronvoluted) [reply docs/hub/spaces.md]: Nice additions. Just made the example suggestion more concise. Will merge when Charles/Christophe has a look!

File: docs/hub/spaces.md
Comment (stevhliu) [inline docs/hub/spaces.md:180]:
```suggestion
Spaces are served in iframes, which by default restrict links from opening in the parent page. The simplest solution is to open them in a new window:
```

File: docs/hub/spaces.md
Comment (stevhliu) [inline docs/hub/spaces.md:186]:
```suggestion
Usually, the height of Spaces is automatically adjusted when using the Gradio library interface. However, if you provide your own frontend in the Gradio SDK and the content height is larger than the viewport, you'll need to add an [iFrame Resizer script](https://cdnjs.com/libraries/iframe-resizer) so the content is scrollable in the iframe:
```

**Review by stevhliu — APPROVED** [review]
Thanks for adding this, I learned something new today! :)

**Review by julien-c — APPROVED** [review]

**Comment by julien-c** [comment]
@ronvoluted After you merge this, you'll need to open a PR against  🌔🛬 to update the `hub-docs` submodule. Let me know if you need any help 👌

File: docs/hub/spaces.md
Comment (ronvoluted) [reply docs/hub/spaces.md:186]: 👍

**Pull request merged**
